src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -436,144 +225,107 @@ def visit_Call(self, node: foast.Call, **kwargs) -> itir.FunCall | itir.Literal:
+        return self.visit(node.args[0], **kwargs)
 
     def _visit_math_built_in(self, node: foast.Call, **kwargs) -> itir.FunCall:
-        args = tuple(to_value(arg)(self.visit(arg, **kwargs)) for arg in node.args)
-        return self._lift_if_field(node)(im.call_(self.visit(node.func, **kwargs))(*args))
+        return self._map(self.visit(node.func, **kwargs), *node.args)
+
+    def _make_reduction_expr(
+        self,
+        node: foast.Call,
+        op: Any,  # TODO fix
+        init_expr: int | itir.Literal,
+        **kwargs,
+    ):
+        # TODO(havogt): deal with nested reductions of the form neighbor_sum(neighbor_sum(field(off1)(off2)))
+        it = self.visit(node.args[0], **kwargs)
+        assert isinstance(node.kwargs["axis"].type, ts.DimensionType)
+        val = im.call_(im.call_("reduce")(op, im.deref_(init_expr)))
+        return im.as_lifted_lambda(val, it)
 
     def _visit_neighbor_sum(self, node: foast.Call, **kwargs) -> itir.FunCall:
-        return self._visit_reduce(node, **kwargs)
+        dtype = type_info.extract_dtype(node.type)
+        return self._make_reduction_expr(node, "plus", self._make_literal("0", dtype), **kwargs)
+
+    def _visit_max_over(self, node: foast.Call, **kwargs) -> itir.FunCall:
+        dtype = type_info.extract_dtype(node.type)
+        min_value, _ = type_info.arithmetic_bounds(dtype)
+        init_expr = self._make_literal(str(min_value), dtype)
+        return self._make_reduction_expr(node, "maximum", init_expr, **kwargs)
+
+    def _visit_min_over(self, node: foast.Call, **kwargs) -> itir.FunCall:
+        dtype = type_info.extract_dtype(node.type)
+        _, max_value = type_info.arithmetic_bounds(dtype)
+        init_expr = self._make_literal(str(max_value), dtype)
+        return self._make_reduction_expr(node, "minimum", init_expr, **kwargs)
 
     def _visit_type_constr(self, node: foast.Call, **kwargs) -> itir.Literal:
         if isinstance(node.args[0], foast.Constant):
             node_kind = self.visit(node.type).kind.name.lower()
             target_type = fbuiltins.BUILTINS[node_kind]
             source_type = {**fbuiltins.BUILTINS, "string": str}[node.args[0].type.__str__().lower()]
             if target_type is bool and source_type is not bool:
-                return im.literal_(str(bool(source_type(node.args[0].value))), node.func.id)
-            return im.literal_(str(node.args[0].value), node_kind)
+                return im.as_lifted_capture(  # TODO seems this codepath is not tested
+                    im.literal_(str(bool(source_type(node.args[0].value))), node.func.id)
+                )
+            return im.as_lifted_capture(im.literal_(str(node.args[0].value), node_kind))
         raise FieldOperatorLoweringError(f"Encountered a type cast, which is not supported: {node}")
 
     def _make_literal(self, val: Any, type_: ts.TypeSpec) -> itir.Literal:
         # TODO(tehrengruber): check constant of this type is supported in iterator ir
```

Comment:
It is not supported in embedded execution, because of an implementation detail: the `offset_provider` is propagated via iterators. If we `lift(lambda: 1.0)()` we will try to extract the offset_provider from the first argument of the lifted lambda (which has no arguments).
However, after we apply all passes, these expressions are inlined...
We should add an executor which does embedded without optimizations to make the pipline cleaner.